src/RequestHandle.cpp

```diff
@@ -389,6 +445,18 @@ auto RequestHandle::setCompletionStatus(
         case CURLcode::CURLE_SSL_CONNECT_ERROR:
             m_status_code = RequestStatus::CONNECT_SSL_ERROR;
             break;
+        case CURLcode::CURLE_WRITE_ERROR:
+            /**
+             * If there is a cURL write error, but the maximum number of bytes
+             * has been written, then we intentionally aborted, so let's set
+             * this to success. Otherwise, it's a legitimate error.
+             */
+            if (GetCompletedWritingMaxBytes()) {
```

Comment:
{ } style, it also seems like these completed writing max bytes functions are internal...